File: apps/node-api/src/routes/health.ts
/** Health and metadata endpoints. Mirrors the Python API's health router. */

import { DISCLAIMER, __version__ as engineVersion } from "@aicodecheck/detector-core";
import { Router } from "express";

import { usingPostgres, usingRedis } from "../config.js";

export const API_VERSION = "0.1.0";
export const router: Router = Router();

router.get("/healthz", (_req, res) => {
  res.json({
    status: "ok",
    api_version: API_VERSION,
    engine_version: engineVersion,
  });
});

router.get("/", (_req, res) => {
  res.json({
    name: "AI Project Detector API",
    runtime: "node",
    api_version: API_VERSION,
    engine_version: engineVersion,
    queue: usingRedis() ? "redis" : "in-process",
    store: usingPostgres() ? "postgres" : "sqlite",
    disclaimer: DISCLAIMER,
  });
});

File: apps/python-api/app/routers/health.py
"""Health and metadata endpoints."""

from __future__ import annotations

from aiprojectdetector import __version__ as engine_version
from aiprojectdetector.models import DISCLAIMER
from fastapi import APIRouter

from .. import __version__ as api_version
from ..jobs import using_redis

router = APIRouter(tags=["meta"])


@router.get("/healthz")
def healthz() -> dict:
    return {"status": "ok", "api_version": api_version, "engine_version": engine_version}


@router.get("/")
def root() -> dict:
    return {
        "name": "AI Project Detector API",
        "api_version": api_version,
        "engine_version": engine_version,
        "queue": "redis" if using_redis() else "in-process",
        "disclaimer": DISCLAIMER,
        "docs": "/docs",
    }
